.github/workflows/build_centos7.yml

```
@@ -262,6 +252,119 @@ jobs:
+    steps:
+      - name: Download build assets
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{needs.build.outputs.TGZ_NAME}}
+
+      - name: Download image
+        uses: actions/download-artifact@v4
+        with:
+          name: centos
+          path: /tmp
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/centos.tar
+
+      - name: setup
+        run: |
+          tar --strip-components=1 -xzf ${{needs.build.outputs.TGZ_NAME}}
+
+      - uses: addnab/docker-run-action@v3
+        name: Run tests
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates --allow-run-as-root
+
+      - uses: addnab/docker-run-action@v3
+        name: run tests with 2 processes
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates -n 2 --allow-run-as-root
+
+  test_single_file_asset:
+    runs-on: ubuntu-latest
+    needs: [ build, build_running_image ]
+    steps:
+      - name: Download build assets
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{needs.build.outputs.SINGLE_FILE_NAME}}
+
+      - name: Download image
+        uses: actions/download-artifact@v4
+        with:
+          name: centos
+          path: /tmp
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/centos.tar
+
+      - name: setup
+        run: |
+          tar --strip-components=1 -xzf ${{needs.build.outputs.SINGLE_FILE_NAME}}
+
+      - uses: addnab/docker-run-action@v3
+        name: Run tests
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates --allow-run-as-root
+
+      - uses: addnab/docker-run-action@v3
+        name: run tests with 2 processes
+        with:
+          image: centos:run
+          shell: bash
+          options: -v ${{ github.workspace }}:/work
+          run: |
+            export PATH=$PATH:/usr/lib64/openmpi/bin
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
+            cd /work
+            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates -n 2 --allow-run-as-root
```

You could merge these two into the same step

```suggestion
      - uses: addnab/docker-run-action@v3
        name: Run tests
        with:
          image: centos:run
          shell: bash
          options: -v ${{ github.workspace }}:/work
          run: |
            export PATH=$PATH:/usr/lib64/openmpi/bin
            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/openmpi/lib
            cd /work
            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates --allow-run-as-root
            ./antares-xpansion-launcher -i examples/SmallTestFiveCandidates -n 2 --allow-run-as-root
```

---

See my comment for justification: https://github.com/AntaresSimulatorTeam/antares-xpansion/pull/898/files#r1743331998